src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state }; // Create a clone of the initial state
+
+  for (const action of actions) {
+    let newState;
+
+    switch (action.type) {
+      case 'addProperties':
+        newState = { ...currentState, ...action.extraData };
```

Comment:
not needed to assign it to `newState`
just change currentstate and push copy in result